core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -579,15 +579,15 @@ package object config {
 
   private[spark] val FILES_MAX_PARTITION_BYTES = ConfigBuilder("spark.files.maxPartitionBytes")
     .doc("The maximum number of bytes to pack into a single partition when reading files.")
-    .longConf
+    .bytesConf(ByteUnit.BYTE)
```

Comment:
I'm not sure this is a net helpful change, as the parameter is maxPartition*Bytes*. I agree it would have been better to call it `maxPartitionSize` and accept values like "10m". I'm not strongly against it, as existing values would still work.

For other property values without "Bytes", I agree.

Reply:
Thanks.
Yeah, the parameter  name is a bit confusing,  but I  think it is not very important whether the parameter name contains "Bytes" or not,  I prefer to change it.

Reply:
I like this change since both styles (i.e. `1024` and `1k`) can be accepted.

Reply:
OK, I'm fine with it.